gobblin-core/src/main/java/gobblin/instrumented/extractor/InstrumentedExtractorDecorator.java

```diff
@@ -80,6 +81,15 @@ public long getHighWatermark() {
   }
 
   @Override
+  public State getFinalState() {
+    if(this.embeddedExtractor instanceof FinalState) {
```

Comment:
You can simply do `return this.getFinalState();` like other `Decorator` classes.